contrib/setup-testbed.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+import argparse
+import subprocess
+import sys
+
+import yaml
+import os
+
+
+def lookup(yaml_path: str, local_data: dict) -> None | str | dict:
+    for bread_crumb in yaml_path.split("."):
+        if bread_crumb in local_data.keys():
+            local_data = local_data[bread_crumb]
+        else:
+            local_data = None
+            break
+    return local_data
+
+
+def check_environment(name: str) -> None:
+    def _check_file(filename: str, name: str):
+        if not os.path.exists(filename):
+            raise RuntimeError(
+                f"There is no file {filename}, create one by using {filename}.sample"
+            )
+
+        with open(filename, "r") as file:
+            cloud_path = f"clouds.{name}"
+            data_local = yaml.safe_load(file)
+            envs = ", ".join(data_local["clouds"].keys())
+
+            if name == "the_environment":
+                raise RuntimeError(
+                    f"you have to specify a name below the 'clouds' element in '{filename}', configured environments are: {envs} (set it with ENVIRONMENT=...)"
+                )
+
+            if lookup(cloud_path, data_local) is None:
+                raise RuntimeError(
+                    f"no such key '{cloud_path}' in '{filename}', configured environments are: {envs} (set it with ENVIRONMENT=...)"
+                )
+
+    try:
+        _check_file("terraform/clouds.yaml", name)
+        _check_file("terraform/secure.yaml", name)
```

Comment:
This file is optional and not required. For example when you work with App Credentials you do normally not have a secure.yaml file.

Reply:
Probably. How about secure.yaml only with a hint without exit 1? If something is missing, it will fail shortly afterwards and you will then see the hint.

The clouds.yaml must always be there, that's ok.

Reply:
A probably typical problem is that the user not enters the appropriate data or the appropriate structure in the secure.yaml on the right place. This causes the process to fail at a later point in time and the user first has to search through the extensive output to realize that he has made this mistake (this is what happened to me). How does the app credential get to Terraform? Is there, for example, an environment variable that I could react to programmatically?

Reply:
Let's make this optional for the moment and we improve it in a followup PR